### DISCUSSION TITLE
edns-client-subnet这个参数，能不能同时设置ipv4和ipv6？

### DISCUSSION THREAD

**Discussion by yyysuo** [pymumu/smartdns#1777] [Q&A]
类似这样
edns-client-subnet 192.168.1.1/24
edns-client-subnet [8::8]/56

问了一下chatgpt，查询AAAA记录时，ecs是v4的地址，也是可以生效的？那就不用再附加v6的地址了吗。smartdns是怎么实现的，查询A只附加ipv4的ecs地址，查询AAAA只附加ipv6的ecs地址吗。

这样配置了一下，没报错
server 223.5.5.5  -subnet ipv4/24 -subnet [ipv6]/60
server 223.6.6.6  -subnet ipv4/24 -subnet [ipv6]/60

**Comment by PikuZheng** [comment]
edns 只是告诉上游dns服务器一个基于ip地址的地理位置，当然ipv4或ipv6都行。
根据我的测试如果同时配置了ipv4和ipv6，会采用ipv4的

**Reply by yyysuo** [reply]
ipv6的sbnet被丢弃了啊......

**Comment by CallMeR** [comment]
之前看到某个 issue 中记录了一个问题，就是某些上游 DNS 服务器并不支持 edns subnet 参数，如果在查询数据包中有这个 subnet 信息，反而会导致上游 DNS 查询失败。  

现在 SmartDNS 能主动探测上游 DNS 是否支持 edns subnet 么？ 如果不行的话，是不是建议自己手动先尝试一下，然后结合实际情况给每个上游 DNS 单独设置（或不设置）edns subnet 参数，也就是不建议使用全局 edns subnet 参数？

**Reply by PikuZheng** [reply]
你说的是check-edns，是smartdns检查上游的edns返回值以判断是不是虚假应答

**Reply by CallMeR** [reply]
那现在建议使用全局的 edns subnet 功能么？ 我单纯用国内的公共 DNS ，然后在全局设置了 edns-client-subnet 和一个所在地附近的 IPv4 地址。

**Reply by PikuZheng** [reply]
> 那现在建议使用全局的 edns subnet 功能么？ 我单纯用国内的公共 DNS ，然后在全局设置了 edns-client-subnet 和一个所在地附近的 IPv4 地址。

两件不同的事吧。我自己是国内国外都不加edns-subnet，因为我自建dns上游和vps在同一台上，没必要加。国内我单一运营商线路也没必要加。如果上游dns是直连公共dns，有必要加（为vps线路优化）。国内没必要，除非是多wan且不同的运营商。

check-edns我认为没必要用。
